larq/metrics.py

```diff
@@ -172,3 +172,60 @@ def get_config(self):
             "values_shape": self.values_shape,
             "values_dtype": self.values_dtype.name,
         }
+
+
+@utils.register_alias("gradient_flow")
+@utils.register_keras_custom_object
+class GradientFlow(LarqMetric):
+    r"""Indicator of gradient mismatch and saturation as described in https://arxiv.org/pdf/1904.02823.pdf.
+        Counts the ratio of activations for each neuron (i.e. pixel in a feature map) over the entire batch
+        that fall in between the HardTanh clipping boundaries ([-1, 1] by default). A neuron with a score of 0
+        did not have any activations in between the boundaries in this batch, and therefore received no gradients 
+        (i.e. saturation). A neuron with a score of 1 was always active in between the boundaries, but therefore
+        suffers from gradient mismatch. Scores are averaged over all neurons in the output.
+
+        # Arguments
+        threshold: The (absolute) clipping threshold used by the HardTanh. 1 by default.
+        name: Name of the metric.
+    """
+
+    def __init__(self, threshold=1.0, name="gradient_flow", dtype=None):
+        super().__init__(name=name, dtype=dtype)
+
+        self.threshold = float(threshold)
+
+        with tf.init_scope():
+            self._threshold = tf.constant(self.threshold, dtype=self.dtype)
+            self.total_value = self.add_weight(
+                "total_value",
+                initializer=tf.keras.initializers.zeros,
+                dtype=self.dtype,
+                aggregation=tf.VariableAggregation.SUM,
+            )
+            self.num_batches = self.add_weight(
+                "num_batches",
+                initializer=tf.keras.initializers.zeros,
+                dtype=tf.int32,
```

Comment:
I think it's fine to store it as float, then we don't need to cast it during run time.
That's how TensorFlow handles it internally: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/metrics.py#L297
```suggestion
```

Reply:
I thought it might be better in terms of performance, since we're adding this integer every batch and only request the float at the end of the epoch.

Reply:
Hmm, might be true. Though I'd stick to what TF is doing internally.

Reply:
@koenhelwegen  any opinions on this?

Reply:
I suppose the performance impact is negligible, since we're only adding 1 number per batch. I'll change it to float then